pkg/scanners/scannerv4/vm_convert.go

```diff
@@ -39,7 +39,26 @@ func toVirtualMachineScanNotes(notes []v4.VulnerabilityReport_Note) []storage.Vi
 	return result
 }
 
+func hasValidCPE(repositories map[string]*v4.Repository, environments map[string]*v4.Environment_List, pkg *v4.Package) bool {
+	envList, ok := environments[pkg.GetId()]
+	if !ok {
+		return false
+	}
+
+	for _, env := range envList.GetEnvironments() {
+		for _, repoID := range env.GetRepositoryIds() {
+			if repositories[repoID].GetCpe() != "" {
+				// A valid CPE is found, therefore the package is scannable.
+				return true
+			}
+		}
+	}
+	return false
+}
+
 func toVirtualMachineComponents(r *v4.VulnerabilityReport) []*storage.EmbeddedVirtualMachineScanComponent {
+	repositories := r.GetContents().GetRepositories()
```

Comment:
It looks like you need to index the repositories by ID between here and the call to `hasValidCPE`